src/reducers/entities/gifs.js

```diff
@@ -0,0 +1,268 @@
+            didInvalidate: true,
+        },
+    }),
+    [GifTypes.REQUEST_SEARCH]: (state, action) => ({
+        ...state,
+        resultsByTerm: TERM_SELECTOR[action.type](state.resultsByTerm, action),
+    }),
+    [GifTypes.RECEIVE_SEARCH]: (state, action) => ({
+        ...state,
+        searchText: action.searchText,
+        resultsByTerm: TERM_SELECTOR[action.type](state.resultsByTerm, action),
+    }),
+    [GifTypes.RECEIVE_SEARCH_END]: (state, action) => ({
+        ...state,
+        searchText: action.searchText,
+        resultsByTerm: TERM_SELECTOR[action.type](state.resultsByTerm, action),
+    }),
+    [GifTypes.RECEIVE_CATEGORY_SEARCH]: (state, action) => ({
+        ...state,
+        searchText: action.searchText,
+        resultsByTerm: TERM_SELECTOR[action.type](state.resultsByTerm, action),
+    }),
+    [GifTypes.SEARCH_FAILURE]: (state, action) => ({
+        ...state,
+        searchText: action.searchText,
+        resultsByTerm: TERM_SELECTOR[action.type](state.resultsByTerm, action),
+    }),
+    [GifTypes.CLEAR_SEARCH_RESULTS]: (state) => ({
+        ...state,
+        searchText: '',
+        resultsByTerm: {},
+    }),
+    [GifTypes.SAVE_SEARCH_SCROLL_POSITION]: (state, action) => ({
+        ...state,
+        scrollPosition: action.scrollPosition,
+    }),
+    [GifTypes.SAVE_SEARCH_PRIOR_LOCATION]: (state, action) => ({
+        ...state,
+        priorLocation: action.priorLocation,
+    }),
+    [GifTypes.UPDATE_SEARCH_TEXT]: (state, action) => ({
+        ...state,
+        searchText: action.searchText,
+    }),
+    [GifTypes.SAVE_SEARCH_BAR_TEXT]: (state, action) => ({
+        ...state,
+        searchBarText: action.searchBarText,
+    }),
+};
+
+const CATEGORIES_SELECTORS = {
+    [GifTypes.REQUEST_CATEGORIES_LIST]: (state) => ({
+        ...state,
+        isFetching: true,
+    }),
+    [GifTypes.CATEGORIES_LIST_RECEIVED]: (state, action) => {
+        const {cursor, tags} = action;
+        const {tagsList: oldTagsList = []} = state;
+        const tagsDict = {};
+        const newTagsList = tags.filter((item) => {
+            if (item && item.gfycats[0] && item.gfycats[0].width) {
+                return true;
+            }
+            return false;
+        }).map((item) => {
+            tagsDict[item.tag] = true;
+            return {
+                tagName: item.tag,
+                gfyId: item.gfycats[0].gfyId,
+            };
+        });
+        const tagsList = [...oldTagsList, ...newTagsList];
+        return {
+            ...state,
+            cursor,
+            hasMore: Boolean(cursor),
+            isFetching: false,
+            tagsList,
+            tagsDict,
+        };
+    },
+    [GifTypes.CATEGORIES_LIST_FAILURE]: (state) => ({
+        ...state,
+        isFetching: false,
+    }),
+};
+
+const TERM_SELECTOR = {
+    [GifTypes.REQUEST_SEARCH]: (state, action) => ({
+        ...state,
+        [action.searchText]: {
```

Comment:
What happens if the user makes a lot of different searches? Will the memory usage of this part of the store ever cause a problem, particularly on mobile where we write the store to disk?

Reply:
If there are a lot of different searches, then this part of the store would grow accordingly with results for each search text until refresh or store is destroyed.  We don't think this is going to be a problem because it just holds text.  Our other integrations have not seen any problems.

Reply:
The store has to be manually destroyed on mobile so we should set a cap on this

Reply:
@enahum What if we just make it so that this isn't persisted to disk on mobile? As long as the app goes into the background at some point, this should be fine

Reply:
Yes this can be done, but it should be done as soon as this PR gets merged